protocol/statetracker/tx_sender.go

```diff
@@ -5,17 +5,102 @@ import (
 
 	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
 	conflicttypes "github.com/lavanet/lava/x/conflict/types"
 )
 
-type TxSender struct{}
+const (
+	defaultGasPrice      = "0.000000001ulava"
+	defaultGasAdjustment = 1.5
+)
+
+type TxSender struct {
+	txFactory tx.Factory
+	clientCtx client.Context
+}
 
-func (ts *TxSender) New(ctx context.Context, txFactory tx.Factory, clientCtx client.Context) (ret *TxSender, err error) {
+func NewTxSender(ctx context.Context, clientCtx client.Context, txFactory tx.Factory) (ret *TxSender, err error) {
 	// set up the rpcClient, and factory necessary to make queries
+	clientCtx.SkipConfirm = true
+	ts := &TxSender{txFactory: txFactory, clientCtx: clientCtx}
+	return ts, nil
+}
+
+func (ts *TxSender) SimulateAndBroadCastTxWithRetryOnSeqMismatch(msg sdk.Msg) error {
```

Comment:
Where is the retry?

Reply:
Maybe add TODO:: add retry mechanism for sequence mismatch
 ;)

Reply:
there is a TODO, and we will do it when we finish rpcprovider